bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/generator/DefaultValueProvider.xtend

```diff
@@ -0,0 +1,192 @@
+			value = ExpressionsFactory.eINSTANCE.createIntLiteral => [
+				value = 0
+			]
+		]
+	}
+	
+	def PrimitiveValueExpression createDummyBool() {
+		return ExpressionsFactory.eINSTANCE.createPrimitiveValueExpression => [
+			value = ExpressionsFactory.eINSTANCE.createBoolLiteral => [
+				value = true
+			]
+		]
+	}
+
+	def dispatch Expression dummyExpression(EnumerationType itemType, EObject context) {
+		if (!itemType.enumerator.isEmpty) {
+			return ExpressionsFactory.eINSTANCE.createElementReferenceExpression => [
+				reference = itemType.enumerator.head
+			]
+		}
+		return createDummyString
+	}
+	def dispatch Expression dummyExpression(StructureType itemType, EObject context) {
+		val args = itemType.parameters.map[p | ExpressionsFactory.eINSTANCE.createArgument => [
+			 value = p.type.dummyExpression(context)
+		]];
+		
+		return ExpressionsFactory.eINSTANCE.createElementReferenceExpression => [
+			reference = itemType;
+			operationCall = true;
+			arguments += args;
+		]
+			
+	}
+	def dispatch Expression dummyExpression(SumType itemType, EObject context) {
+		if (!itemType.alternatives.isEmpty) {
+			val alt = itemType.alternatives.head;
+			val args = alt.accessorsTypes.map[tp | ExpressionsFactory.eINSTANCE.createArgument => [
+				 value = tp.dummyExpression(context)
+			]];
+			if(EcoreUtil2.getContainerOfType(context, SystemResourceSetup) === null) {
+				return ExpressionsFactory.eINSTANCE.createFeatureCall => [
+					feature = alt;
+					operationCall = true;
+					owner = ExpressionsFactory.eINSTANCE.createElementReferenceExpression => [
+						reference = itemType;
+					]
+					arguments += args;
+				]
+			}
+			return ExpressionsFactory.eINSTANCE.createElementReferenceExpression => [
+				reference = itemType.alternatives.head;
+				operationCall = true;
+				arguments += args;
+			]
+			
+		}
+		return createDummyString
+	}
+
+	def dispatch dummyExpression(AbstractSystemResource itemType, EObject context) {
+		val scope = getSetupScope(itemType, context);
+		if (!scope.allElements.empty) {
+			return ExpressionsFactory.eINSTANCE.createElementReferenceExpression => [
+				reference = scope.allElements.head.EObjectOrProxy
+			]
+		}
+		return createDummyString
+	}
+	
+	protected def FilteringScope getSetupScope(AbstractSystemResource itemType, EObject context) {
+		new FilteringScope(
+			scopeProvider.getScope(context, ExpressionsPackage.Literals.ELEMENT_REFERENCE_EXPRESSION__REFERENCE),
+			[
+				ProgramPackage.Literals.SYSTEM_RESOURCE_SETUP.isSuperTypeOf(it.EClass) &&
+					(it.EObjectOrProxy as SystemResourceSetup).type == itemType
+			]
+		)
+	}
+	
+	def String getDummyConstructor(String base, ComplexType typ) {
+		val namedParamsOpt = ModelUtils.getAccessorParameters(typ);
+		if(namedParamsOpt.present) {
+			val namedParams = namedParamsOpt.get
+			val proposalString = '''«base»«typ.name»(«FOR param : namedParams SEPARATOR(", ")»«param.name» = «getDummyString(param.type)»«ENDFOR»)'''
```

Comment:
There should not be a space between the parameter name and the value, i.e. "foo=123" rather than "foo = 123"

Reply:
shouldn't there? I've read some R where this is also common practice and I find it harder to read.